pipelines/batch/README.md

```diff
@@ -0,0 +1,193 @@
+
+## Setup
+
+1.  [Clone](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository)
+    the [FHIR Data Pipes](https://github.com/google/fhir-data-pipes) project to
+    your machine.
+1.  Set the `utils` directory to world-readable: `chmod -R 755 ./utils`.
+1.  Build binaries by running `mvn clean install` from the root directory of the
+    repository.
+
+## Run the pipeline
+
+Run the pipeline directly using the `java` command:
+
+```
+java -cp ./pipelines/batch/target/batch-bundled-0.1.0-SNAPSHOT.jar \
+    org.openmrs.analytics.FhirEtl \
+    --fhirServerUrl=http://example.org/fhir \
+    --[see additional parameters below]
+```
+
+Add the necessary parameters depending on your use case. The methods used for
+reading the source FHIR server and outputting the data depend on the parameters
+used. You can output to both Parquet files and a FHIR server by including the
+required parameters for both.
+
+## Parameters
+
+This section documents the parameters used by the various pipelines. For more
+information on parameters, see
+[`FhirEtlOptions`](https://github.com/google/fhir-data-pipes/blob/master/pipelines/batch/src/main/java/org/openmrs/analytics/FhirEtlOptions.java).
+
+### Common parameters
+
+These parameters are used regardless of other pipeline options.
+
+- `fhirServerUrl` - The base URL of the source FHIR server. Required.
+- `fhirServerUserName` - The HTTP Basic Auth username to access the FHIR server
+  APIs. Default: `admin`
+- `fhirServerPassword` - The HTTP Basic Auth password to access the FHIR server
+  APIs. Default: `Admin123`
+- `resourceList` - A comma-separated list of
+  [FHIR resources](https://www.hl7.org/fhir/resourcelist.html) to include in the
+  pipeline. Default: `Patient,Encounter,Observation`
+- `runner` -
+  [The Apache Beam Runner](https://beam.apache.org/documentation/runners/capability-matrix/)
+  to use. Pipelines supports DirectRunner and FlinkRunner by default; other
+  runners can be enabled by Maven profiles, e.g.,
+  [DataflowRunner](https://github.com/google/fhir-data-pipes/blob/16fcc255cef4d2708b9941a854e6c638b2533d45/pipelines/batch/pom.xml#L257).
+  Default: `DirectRunner`
+
+### FHIR-Search input parameters
+
+The pipeline will use FHIR-Search to fetch data as long as `jdbcModeEnabled` is
+unset or false.
+
+- `batchSize` - The number of resources to fetch in each API call. Default:
+  `100`
+
+### JDBC input parameters
+
+JDBC mode is used if `jdbcModeEnabled=true`.
+
+To use JDBC mode, first create a copy of
+[hapi-postgres-config.json](https://github.com/google/fhir-data-pipes/blob/master/utils/hapi-postgres-config.json)
+and edit the values to match your database server.
+
+Next, include the following parameters:
+
+- `jdbcModeEnabled=true`
+- `fhirDatabaseConfigPath=./path/to/config.json`
+
+If you are using a HAPI FHIR server, also include:
+
+- `jdbcModeHapi=true`
+- `jdbcDriverClass=org.postgresql.Driver`
```

Comment:
As mentioned above, this is not really specific to HAPI and can apply to OpenMRS too. BTW, I am thinking that maybe we should change the default value of this to be for PostgreSQL not MySQL; WDYT?

Reply:
Changing the default sounds good to me.